modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -566,4 +554,40 @@ TEST(Objdetect_QRCode_Encode_Decode, auto_version_pick)
     }
 }
 
+// Test two QR codes which error correction procedure requires more number of
+// syndroms that described in the ISO/IEC 18004
+TEST(Objdetect_QRCode_decode, error_correction)
+{
+    const std::string filename[] = {"err_correct_1M.png", "err_correct_2L.png"};
+    const std::string expected[] = {"New", "Version 2 QR Code Test Image"};
+
+    QRCodeDetector qrcode;
+    cv::String decoded_msg;
+    for (int i = 0; i < 2; ++i)
+    {
```

Comment:
I propose to implement it as parameterized test. In case of failure the case is visible in logs.